client/asset/eth/eth.go

```diff
@@ -214,17 +221,15 @@ type ethFetcher interface {
 	transactionConfirmations(context.Context, common.Hash) (uint32, error)
 	txOpts(ctx context.Context, val, maxGas uint64, maxFeeRate *big.Int) (*bind.TransactOpts, error)
 	currentFees(ctx context.Context) (baseFees, tipCap *big.Int, err error)
+	unlock(pw string) error
 }
 
-// Check that ExchangeWallet satisfies the asset.Wallet interface.
-var _ asset.Wallet = (*ExchangeWallet)(nil)
-var _ asset.AccountLocker = (*ExchangeWallet)(nil)
-var _ asset.FeeRater = (*ExchangeWallet)(nil)
+// Check that AssetWallet satisfies the asset.Wallet interface.
+var _ asset.Wallet = (*AssetWallet)(nil)
+var _ asset.AccountLocker = (*AssetWallet)(nil)
+var _ asset.TokenMaster = (*AssetWallet)(nil)
 
-// ExchangeWallet is a wallet backend for Ethereum. The backend is how the DEX
-// client app communicates with the Ethereum blockchain and wallet. ExchangeWallet
-// satisfies the dex.Wallet interface.
-type ExchangeWallet struct {
+type baseWallet struct {
```

Comment:
There is only one `baseWallet` shared by all `AssetWallet`s. 

https://github.com/decred/dcrdex/blob/64b5c1e55e9a75937c4bd3d4f523bda17fff3813/client/asset/eth/eth.go#L512

I am going to look at `ETHWallet` and `TokenWallet` structs as per @JoeGruffins request though.

Reply:
> There is only one `baseWallet` shared by all `AssetWallet`s.

Yah, this was really what I was commenting about.  Made me uncertain on why there where two separate types.  Will see what happens with the next iteration.

Reply:
It's because there's only one instance of `baseWallet` that holds the reference to the node. Then there are multiple  instances of `AssetWallet` each of which hold a reference to the one `baseWallet`. It's not that each `AssetWallet` is holding a reference to it's own `baseWallet`.